llvm-external-projects/iree-dialects/lib/Dialect/LinalgExt/IR/LinalgExtOps.cpp

```diff
@@ -1265,6 +1265,126 @@ LogicalResult TopkOp::verify() {
   return success();
 }
 
+SmallVector<Range> TopkOp::getIterationDomain(OpBuilder &builder) {
+  int64_t operandRank = getInputRank();
+  SmallVector<Range> loopBounds(operandRank);
+  Location loc = getLoc();
+  Value zero = builder.create<arith::ConstantIndexOp>(loc, 0);
+  Value one = builder.create<arith::ConstantIndexOp>(loc, 1);
+  Value source = values();
+
+  for (auto dim : llvm::enumerate(getInputType().getShape())) {
+    loopBounds[dim.index()].offset = zero;
+    if (dim.value() == ShapedType::kDynamicSize) {
+      loopBounds[dim.index()].size =
+          getDimValue(builder, loc, source, dim.value());
+    } else {
+      loopBounds[dim.index()].size =
+          builder.create<arith::ConstantIndexOp>(loc, dim.value());
+    }
```

Comment:
I think we can use `getDim` method. It's defined in l.101.

```cpp
  OpFoldResult IREE::LinalgExt::getDim(OpBuilder &builder, Location loc, Value v,
                                       int64_t dim) {
    auto t = v.getType().cast<ShapedType>();
    if (t.isDynamicDim(dim)) {
      return getDimValue(builder, loc, v, dim);
    }
    return builder.getI64IntegerAttr(t.getDimSize(dim));
  }
```

Reply:
Similar function. But `builder.getI64IntegerAttr(t.getDimSize(dim));` returns an Attribute while we can only handle `Value` types here. I don't see another method that provides this by default so I'd stick to this. 

The other implementations of `getIterationDomain` don't handle the constant case separately (only use `getDimValue`). I did this for both ease of debugging/lowering and simpler IR. Could look at the others but would prefer separate PR.